firmware/chipignite/util/caravel/hk.py

```diff
@@ -0,0 +1,155 @@
+            print("status reg_1 = {}".format(hex(self.get_status())))
+        else:
+            print("status reg_1 = {}".format(hex(self.get_status())))
+            status = self.slave.exchange([CARAVEL_PASSTHRU, 0x35], 1)
+            print("status reg_2 = {}".format(hex(int.from_bytes(status, byteorder='big'))))
+            # print("status = {}".format(hex(from_bytes(slave.exchange([CMD_READ_STATUS], 2)[1], byteorder='big'))))
+
+
+
+    def is_busy(self):
+        return self.get_status() & SR_WIP
+
+    def read_reg(self, reg: int):
+        return self.slave.exchange([CARAVEL_REG_READ, reg], 1)
+    
+    def write_reg(self, reg: int, value: int):
+        self.slave.exchange([CARAVEL_REG_WRITE, reg, value], 1)
+
+    def read_project_id(self):
+        data = self.slave.exchange([CARAVEL_STREAM_READ, 0x04], 4)
+        return int('{:032b}'.format(int.from_bytes(data, byteorder='big'))[::-1], 2)
+
+
+
+
+    def cpu_reset_hold(self):
+        self.slave.write([CARAVEL_REG_WRITE, 0x0b, 0x01])
+
+    def cpu_reset_release(self):
+        self.slave.write([CARAVEL_REG_WRITE, 0x0b, 0x00])
+
+    def cpu_reset_toggle(self):
+        self.cpu_reset_hold()
+        self.cpu_reset_release()
+
+
+
+    def flash_reset(self):
+        self.slave.write([CARAVEL_PASSTHRU, CMD_RESET_CHIP])
+        self.print_status()
+
+    def flash_read_jedec(self):
+        return self.slave.exchange([CARAVEL_PASSTHRU, CMD_JEDEC_DATA], 3)
+    
+    def flash_identify(self):
+        jedec = self.flash_read_jedec()
+        print("JEDEC = {:06x}".format(int.from_bytes(jedec, byteorder='big')))
+
+        if jedec[0:1] != bytes.fromhex('ef'):
+        # if jedec[0:1] != bytes.fromhex('e6'):
+            print("Winbond SRAM not found")
+            exit(1)
+
+    def flash_write_enable(self):
+        self.slave.write([CARAVEL_PASSTHRU, CMD_WRITE_ENABLE])
+
+    def flash_erase(self, wait=True, quiet=False):
+        if not quiet: print("Erasing chip...")
+
+        self.flash_write_enable()
+        self.slave.write([CARAVEL_PASSTHRU, CMD_ERASE_CHIP])
+
+        if wait:
+            while (self.is_busy()):
+                time.sleep(0.3)
+                if not quiet: print('.', end='', flush=True)
+                self.led.toggle()
+
+            if not quiet: print("done")
+            if not quiet: self.print_status()
+    
+
+
+
+    def engage_dll(self):
+        self.slave.write([CARAVEL_REG_WRITE, 0x08, 0x01])
+        self.slave.write([CARAVEL_REG_WRITE, 0x09, 0x00])
+
+    def read_dll_trim(self):
+        return self.slave.exchange([CARAVEL_STREAM_READ, 0x0d], 4)
+    
+    def disengage_dll(self):
+        self.slave.write([CARAVEL_REG_WRITE, 0x09, 0x01])
+        self.slave.write([CARAVEL_REG_WRITE, 0x08, 0x00])
+    
+    def dco_mode(self):
+        self.slave.write([CARAVEL_REG_WRITE, 0x08, 0x03])
+        self.slave.write([CARAVEL_REG_WRITE, 0x09, 0x00])
+
+    def dco_trim(self, value: (int,int,int,int)):
```

Comment:
The tuple here (at least, with respect to the lines below which write into registers) is in little-endian order (i.e. `( [7:0], [15:8], [23:16], [25:24] )`) -- would it be better to reverse these? Either way, a comment should be included to make it clear what order the bit packing is for this argument.